pkg/kubelet/dockershim/naming.go

```diff
@@ -92,7 +102,9 @@ func parseSandboxName(name string) (*runtimeapi.PodSandboxMetadata, error) {
 	name = strings.TrimPrefix(name, "/")
 
 	parts := strings.Split(name, nameDelimiter)
-	if len(parts) != 6 {
+	// Tolerate the random suffix.
+	// TODO(random-liu): Remove 7 field case when docker 1.11 is deprecated.
+	if len(parts) != 6 && len(parts) != 7 {
```

Comment:
I agree that this should not happen. Just wanted to know what the behavior would be in case it happens due to buggy code. I think kubelet sorts the containers by creation timestamp and go through the latest one first, so this should not be an issue anyway.